### PR TITLE
clean and refactor ngenius service

### PR DESCRIPTION
**Pull request by DFSko**
resolves happy-travel/agent-app-project#662

### CODE REVIEW

File: Api/Extensions/AgentAgencyExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using HappyTravel.Edo.Api.Models.Agencies;
+using HappyTravel.Edo.Api.Models.Agents;
+using HappyTravel.Edo.Api.Models.Payments.NGenius;
+
+namespace HappyTravel.Edo.Api.Extensions
```
Comment (mingazhev) [inline Api/Extensions/AgentAgencyExtensions.cs:5]: Please don't place such extensions to this common folder. Extension is used in one or two place, move it to namespaces nearer to usage, may be to the same folder with service where it is used

File: Api/Extensions/MoneyAmountExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using HappyTravel.Edo.Api.Models.Payments.NGenius;
+using HappyTravel.Money.Extensions;
+using HappyTravel.Money.Models;
+
+namespace HappyTravel.Edo.Api.Extensions
+{
+    public static class MoneyAmountExtensions
```
Comment (mingazhev) [inline Api/Extensions/MoneyAmountExtensions.cs:7]: Same there, nearer to NGenius services

File: Api/Services/Payments/PaymentService.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+using HappyTravel.Edo.Common.Enums;
+using HappyTravel.Edo.Data;
+using HappyTravel.Edo.Data.Payments;
+using HappyTravel.Money.Models;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+
+namespace HappyTravel.Edo.Api.Services.Payments
+{
+    public class PaymentService : IPaymentService
```
Comment (mingazhev) [inline Api/Services/Payments/PaymentService.cs:16]: PaymentsManagementService maybe?
Or PaymentRecordsService?

`PaymentService` sounds like the God-service responsible for all the payments

Comment (mingazhev) [reply Api/Services/Payments/PaymentService.cs:16]: And as I see it has credit-card specific things. This should be reflected in the name, if this service is used for credit card payments management

File: Api/Extensions/AgentAgencyExtensions.cs (hunk 1 above)
Comment (DFSko) [reply Api/Extensions/AgentAgencyExtensions.cs]: Fixed

File: Api/Extensions/MoneyAmountExtensions.cs (hunk 2 above)
Comment (DFSko) [reply Api/Extensions/MoneyAmountExtensions.cs]: Fixed

File: Api/Services/Payments/PaymentService.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Payments/PaymentService.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**